File: Pbm/Codeforces/2026.1-2026.3/2199B.cpp
//2026/3/21
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define endl '\n'

void solve() {
  int a, b, c, d; cin >> a >> b >> c >> d;
  cout << c+d-a-b-max(0, min(c, d)-max(a, b)) << endl;
}

int main() {
  cin.tie(0) -> sync_with_stdio(0);
  int T = 1; cin >> T;
  while(T--) solve();
  return 0;
}

File: Pbm/Codeforces/2026.1-2026.3/2199B.kt
import kotlin.math.max
import kotlin.math.min

fun solve() {
    // 读取四个整数，与C++的 `cin >> a >> b >> c >> d` 对应
    val (a, b, c, d) = readln().split(" ").map { it.toInt() }
    // 计算公式: c + d - a - b - max(0, min(c, d) - max(a, b))
    val result = c + d - a - b - max(0, min(c, d) - max(a, b))
    println(result)
}

fun main() {
    // 读取整数T，对应C++的 `int T; cin >> T;`
    var T = readln().toInt()
    // 循环T次执行solve函数
    while (T-- > 0) {
        solve()
    }
}